src/deploy/functions/tasks.ts

```diff
@@ -94,7 +95,10 @@ export function createFunctionTask(
       }
       op = await gcf.createFunction(apiFunction);
     } else {
-      const apiFunction = await gcfV2.functionFromSpec(fn, params.storage![fn.region]);
+      if (backend.isEventTrigger(fn.trigger) && !fn.trigger.region) {
+        await setTriggerRegionFromTriggerType(fn.trigger);
```

Comment:
Let's do simple lookups before the release phase. I'd rather these errors be discovered before we start changing the customer's backend.

Reply:
I believe the best place to perform this lookup is in the prepare step. Since we are already looping through each functionSpec and checking for a cached value, performing the API lookup there seems like the best place.